server/src/test/java/org/elasticsearch/search/SearchServiceTests.java

```diff
@@ -1346,34 +1346,34 @@ public void testWaitOnRefresh() throws Exception {
         final IndexShard indexShard = indexService.getShard(0);
         SearchRequest searchRequest = new SearchRequest().allowPartialSearchResults(true);
 
+        final IndexResponse response = client().prepareIndex("index").setSource("id", "1").get();
+        assertEquals(RestStatus.CREATED, response.status());
+
         SearchShardTask task = new SearchShardTask(123L, "", "", "", null, Collections.emptyMap());
         PlainActionFuture<SearchPhaseResult> future = PlainActionFuture.newFuture();
         ShardSearchRequest request = new ShardSearchRequest(OriginalIndices.NONE, searchRequest, indexShard.shardId(), 0, 1,
-            new AliasFilter(null, Strings.EMPTY_ARRAY), 1.0f, -1, null, null, null, 0);
-        request.source(SearchSourceBuilder.searchSource().timeout(TimeValue.timeValueSeconds(30)));
+            new AliasFilter(null, Strings.EMPTY_ARRAY), 1.0f, -1, null, null, null, 0, TimeValue.timeValueSeconds(30));
         service.executeQueryPhase(request, task, future);
-
-        final IndexResponse response = client().prepareIndex("index").setSource("id", "1").get();
-        assertEquals(RestStatus.CREATED, response.status());
-        searchRequest.indices("index").source(new SearchSourceBuilder().query(new TermQueryBuilder("id", "1")));
-
         SearchPhaseResult searchPhaseResult = future.actionGet();
         assertEquals(1, searchPhaseResult.queryResult().getTotalHits().value);
     }
 
```

Comment:
Also add a test that searching against an index with disabled refresh using `waitForCheckpoints` fails.